### PR TITLE
Release updates for grid: Update version and dependents

### PR DESCRIPTION
**Pull request by bjankord**
Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-site/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "terra-clinical-item-view": "^0.x",
     "terra-content": "^0.x",
     "terra-detail-view": "^0.x",
-    "terra-grid": "^1.0.0",
+    "terra-grid": "^2.0.0",
```
Comment (StephenEsser) [inline packages/terra-site/package.json]: The terra-detail-view still has a DevDependency on terra-grid, this was used for examples before they moved to terra-site, but I think there are still tests utilizing the grid. Should terra-detail-view also be bumped, or is it something we can just update in the future?

Comment (bjankord) [inline packages/terra-site/package.json]: Good call. Versioning is a bit odd here. `terra-grid v1.0.0` does not contain react components. So while `terra-detail-view` denotes a dev dependency on `terra-grid: ^1.0.0` it was really referencing the code in terra-grid that would become `terra-grid v2.0.0` via npm-link. I'm open to talking about this more in our bi-weekly meetings. There is some functionality around our release script that led us to using this strategy.

**Comment by Matt-Butler** [comment]
Just curious if this is an issue yet? https://github.com/cerner/terra-core/issues/4

**Comment by bjankord** [comment]
@Matt-Butler #4 is still an issue. Fill land bug fix for it in a future release.

**Pull request merged**